### DISCUSSION TITLE
Pasted Code Limitations?

### DISCUSSION THREAD

**Discussion by JimNayzium** [community/community#72830] [Copilot Conversations]
### Select Topic Area

General

### Body

I have found Copilot to be incredible as long as the code I paste into the Chat is short enough to not throw a "Code too long" message. However, sometimes Copilot is spooky like C3P0 and will tell me what is wrong with the code ON MY open editor page, (or at least it feels that way, I guess I could have pasted that code in hours ago and it somehow remembered).

My question is, is there anyway to GRANT VSCODE Copilot permission to read my actual open editor window and spare me having to figure out how to break down the code and paste it in to the chat? 

Sometimes I just want Copilot to read over the entire javascript class and tell me what is redundant or where and how to fix the vscode obsure error type thing.

Is this possible?

**Comment by abdulrahim-ramadan** [comment]
Response#1
